### PR TITLE
Added common modules for python topology support

### PR DESCRIPTION
**Pull request by taishi8117**
This PR includes basic common modules necessary for adding python topology support

Related PR: #1136, #1137, #1138

This PR depends on #1150.

### CODE REVIEW

File: heron/common/src/python/constants.py [hunk 1]
```diff
@@ -25,6 +28,32 @@
...
+
 RAM_FOR_STMGR = 1 * GB
 DEFAULT_RAM_FOR_INSTANCE = 1 * GB
 DEFAULT_DISK_PADDING_PER_CONTAINER = 12 * GB
+
+LOGGING_DIRECTORY = 'heron.logging.directory'
+MAX_LOG_FILES = 'heron.logging.maximum.files'
+MAX_LOG_SIZE_MB = 'heron.logging.maximum.size.mb'
+
+INSTANCE_ACK_BATCH_TIME_MS = "heron.instance.ack.batch.time.ms"
```
Comment (kramasamy) [inline heron/common/src/python/constants.py]: There were a bunch of other heron instance options. Wondering if they are not needed?

**Comment by kramasamy** [comment]
@taishi8117 - a few feedback.

File: heron/common/src/python/basics/event_looper.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+'''event_looper.py'''
+import time
+import traceback
+import sys
+
+from abc import abstractmethod
+from heapq import heappush, heappop
+from heron.common.src.python.log import Log
+
+class EventLooper(object):
```
Comment (billonahill) [inline heron/common/src/python/basics/event_looper.py]: could you add a description of what the EventLooper is and how it should be used? Same for GatewayLooper.

File: heron/common/src/python/basics/event_looper.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  """Event Looper can block the thread on ``do_wait()`` and unblock on ``wake_up()``"""
+  def __init__(self):
+    self.should_exit = False
+    self.wakeup_tasks = []
+    self.timer_tasks = []
```
Comment (billonahill) [inline heron/common/src/python/basics/event_looper.py]: This code isn't thread safe. Is it not intended to be invoked by multiple threads? If not it should be documented as such. If so, these objects should be accesses in a thread safe manner.

File: heron/common/src/python/constants.py (hunk 1 above)
Comment (taishi8117) [inline heron/common/src/python/constants.py]: I opened another PR (#1150) for just constants.py.

File: heron/common/src/python/basics/event_looper.py (hunk 3 above)
Comment (taishi8117) [inline heron/common/src/python/basics/event_looper.py]: @billonahill Yes, this code is not intended to be thread safe, as it is not for Wakeable Looper. I will add documentation for that.

**Comment by kramasamy** [comment]
👍 @billonahill - do you have any other feedback?

File: heron/common/src/python/basics/event_looper.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+import sys
+
+from abc import abstractmethod
+from heapq import heappush, heappop
+from heron.common.src.python.log import Log
+
+class EventLooper(object):
+  """EventLooper is a Python implementation of WakeableLooper.java
+
+  EventLooper is a generic class that could:
```
Comment (billonahill) [inline heron/common/src/python/basics/event_looper.py]: should clarify that it's a class for scheduling recurring tasks.

File: heron/common/src/python/basics/event_looper.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+    This will start a while loop until ``exit_loop()`` is called.
+    """
+    while not self.should_exit:
+      self.run_once()
+
+    self.on_exit()
+
+  def run_once(self):
+    """Run once, should be called only from loop()"""
```
Comment (billonahill) [inline heron/common/src/python/basics/event_looper.py]: should this be private if only loop should call it?

File: heron/common/src/python/basics/event_looper.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    second_in_float = float(second)
+    expiration = time.time() + second_in_float
+    heappush(self.timer_tasks, (expiration, task))
+
+  def exit_loop(self):
+    """Exits the loop"""
+    self.should_exit = True
+    self.wake_up()
+
+  def get_next_timeout_interval(self):
```
Comment (billonahill) [inline heron/common/src/python/basics/event_looper.py]: should this method and the ones below be private?

File: heron/common/src/python/basics/gateway_looper.py [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+    # Pipe used for wake up select
+    self.pipe_r, self.pipe_w = os.pipe()
+
+    self.started = time.time()
+    Log.debug("Gateway Looper started time: " + str(time.asctime()))
+
+  def prepare_map(self, sock_map):
+    """Specifies which socket map to use; should be asyncore's socket map"""
+    self.sock_map = sock_map
```
Comment (billonahill) [inline heron/common/src/python/basics/gateway_looper.py]: Could this be done in the constructor? How come a separate method to set the map?

File: heron/common/src/python/basics/gateway_looper.py [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    os.close(self.pipe_w)
+
+  # pylint: disable=too-many-branches
+  def poll(self, timeout=0.0):
+    """Modified version of poll() from asyncore module"""
+    if self.sock_map is None:
+      Log.warning("Socket map is not registered to Gateway Looper")
+    r = []
+    w = []
+    e = []
```
Comment (billonahill) [inline heron/common/src/python/basics/gateway_looper.py]: better to use descriptive member names than single letters.

**Comment by billonahill** [comment]
👍 for the change but please check on the CI failure before shipping.

**Pull request merged**